var/spack/repos/builtin/packages/m4/package.py

```diff
@@ -39,6 +39,16 @@ class M4(Package):
 
     depends_on('libsigsegv', when='+sigsegv')
 
+    def _make_executable(self, name):
+        return Executable(join_path(self.prefix.bin, name))
+
+    def setup_dependent_package(self, module, dependent_spec):
+        # m4 is very likely to be a build dependency, so we add the
```

Comment:
From what I understand, this block of code is only needed if you want to run that executable in another package's install method. For example, the `autoconf` package declares a method called `autoreconf()` which runs the `autoreconf` executable when it is called. Now, other packages that depend on `autoconf` can run `autoreconf()` to rebuild their configure script. So, unless someone needs to call `m4()` in their package, this isn't really necessary.